workitem/number_sequence/workitem_number_sequence_repository.go

```diff
@@ -0,0 +1,56 @@
+package numbersequence
+
+import (
+	"context"
+
+	"github.com/fabric8-services/fabric8-wit/log"
+	"github.com/jinzhu/gorm"
+	errs "github.com/pkg/errors"
+	uuid "github.com/satori/go.uuid"
+)
+
+// WorkItemNumberSequenceRepository the interface for the work item number sequence repository
+type WorkItemNumberSequenceRepository interface {
+	Create(ctx context.Context, spaceID uuid.UUID) (*WorkItemNumberSequence, error)
```

Comment:
Do we care about create? You have all the info on nextVal to create it

Reply:
actually, I think it's worth creating the "sequence" upon space creation, so we can effectively lock the row in the table when adding a work item. I suspect that this is what causes the `duplicate key` issue during the tests.